### DISCUSSION TITLE
​[Usability enhancement request] Explicit selection for indentation target

### DISCUSSION THREAD

**Discussion by houshasen** [toeverything/AFFiNE#6959] [Ideas]
### Description

I find current indent/unindent slightly counterintuitive, and often result in extra steps.

**Current Behavior:** When indenting an item, all underlying blocks automatically indent as well. Similarly, deleting a line with lower-level indents causes all underlying items to unindent. This results in more steps compared to other editors when adding a new indented item or deleting a single indented line.

https://github.com/toeverything/AFFiNE/assets/48835895/55b0293c-a242-4042-a3a3-893257354108

**Suggestion:** Make the indent/unindent selection explicit. If a user wants to shift all items below the current line, they should select the blocks and use tab or shift-tab. If only a single line is selected and tab or shift-tab, it should indent/unident just that line.

Below is how Notion works.

https://github.com/toeverything/AFFiNE/assets/48835895/03b78d8a-7de2-433e-a737-cd7853cb29ed

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

[a comment by an automated account is not shown]

**Comment by CatsJuice** [comment]
This seems to be designed early on, I will turn it into a discussion, and then plan how to adjust it.
